src/pages/ItemListPage/index.js

```diff
@@ -1,23 +1,28 @@
 import React, { useEffect, useState } from 'react'
 import { ItemList } from '../../components/ItemList'
 import './ItemListPage.css'
+import { SearchField } from '../../components/SearchField'
+import { useLocation } from 'react-router-dom'
 
-export const ItemListPage = () => {
+export const ItemListPage = props => {
   const [items, setItems] = useState([])
-
+  const query = new URLSearchParams(useLocation().search)
   useEffect(() => {
     async function fetchItems() {
-      const response = await fetch('http://localhost:4000/items', {
+      let targetUrl = `http://localhost:4000/items?name=${query.get('name') ||
+        ''}&category=${query.get('category') || ''}`
+      const response = await fetch(targetUrl, {
         method: 'GET',
       })
       const items = await response.json()
       setItems(items)
     }
     fetchItems().catch(e => console.error(e))
-  }, [])
+  })
```

Comment:
```suggestion
  }, [])
```
Without dependencies here this is causing infinite calls.